modules/csv/csv.go

```diff
@@ -102,3 +129,20 @@ func FormatError(err error, locale translation.Locale) (string, error) {
 
 	return "", err
 }
+
+// Looks for possible delimiters right before or after (with spaces after the former) double quotes with closing quotes
+var beforeAfterQuotes = regexp.MustCompile(`([,@\t;|]{0,1}) *(?:"[^"]*")+([,@\t;|]{0,1})`)
+
+// guessFromBeforeAfterQuotes guesses the limiter by finding a double quote that has a valid delimiter before it and a closing quote,
+// or a double quote with a closing quote and a valid delimiter after it
+func guessFromBeforeAfterQuotes(data []byte) rune {
+	rs := beforeAfterQuotes.FindStringSubmatch(string(data))
+	if rs != nil {
+		if rs[1] != "" {
+			return rune(rs[1][0])
+		} else if rs[2] != "" {
+			return rune(rs[2][0])
+		}
```

Comment:
Can you guarantee that there will be three array elements?

Reply:
Yes. if it isn't nil (no match) then it comes back with rs[0] the whole match, rs[1] the first set of parentheses (a delimiter left of quoted string), rs[2] the second set of parentheses (a delmiter right of quoted string). There are tests for this for when there is no match, a left match, a right match, or both

Reply:
Added some comments to clarify what is going on.